app_flutter/lib/service/appengine_cocoon.dart

```diff
@@ -86,22 +86,31 @@ class AppEngineCocoonService implements CocoonService {
   @override
   Future<bool> rerunTask(Task task, String accessToken) async {
     assert(accessToken != null);
+    final String postResetTaskUrl = _apiEndpoint('/api/reset-devicelab-task');
 
     /// This endpoint only returns a status code.
-    final http.Request request = http.Request(
-        'POST', Uri.https(_appengineAuthority, '/api/reset-devicelab-task'))
-      ..bodyFields = <String, String>{
-        'Key': task.key.toString(),
-      }
-      ..headers.addAll(<String, String>{
-        'X-Flutter-AccessToken': accessToken,
-      });
-
-    final http.StreamedResponse response = await _client.send(request);
+    final http.Response response =
+        await _client.post(postResetTaskUrl, headers: <String, String>{
+      'X-Flutter-AccessToken': accessToken,
+    }, body: <String, String>{
+      'Key': task.key.toString(),
+    });
 
     return response.statusCode == HttpStatus.ok;
   }
 
+  /// Construct the API endpoint based on the priority of using a local endpoint
+  /// before falling back to the production endpoint.
+  ///
+  /// This function helps the Web app talks to the Cocoon service relative to
+  /// where it's running. On mobile, default to use the production
```

Comment:
Explain the magic: "On Web, a relative url automatically resolves to a local Cocoon instance when available, otherwise it resolvs to the production one. Therefore this function returns a relative url for Web.

Mobile apps don't support relative url."

Reply:
Okay. I expanded to explain more on the relative url, and why we do this.